File: code/angular/src/app/model3/models/Model3Year.ts
export class Model3Year {
  calendarYear!: number;
  age!: number;
  yearIndex!: number;
  priorYear!: number;
  amountAtStart!: number;
  investmentReturn!: number;
  allocations!: Model3AllocationYear[];
  amountAtEnd!: number;
}

export class Model3AllocationYear {
  name!: string;
  amountAtStart!: number;
  investmentReturn!: number;
}

File: code/dotnet/PredictorLibrary/Model3/Outputs/Model3Year.cs
namespace PredictorLibrary.Model3.Outputs;

public class Model3Year
{
    public int CalendarYear { get; set; }
    public int Age { get; set; }
    public int YearIndex { get; set; }
    public decimal PriorYear { get; set; }
    public decimal AmountAtStart { get; set; }

    public decimal InvestmentReturn
    {
        get { return Allocations.Sum(allocation => allocation.InvestmentReturn); }
    }

    public IList<Model3AllocationYear> Allocations { get; set; } = new List<Model3AllocationYear>();

    public decimal AmountAtEnd => AmountAtStart + InvestmentReturn;
}